src/index.ts

```diff
@@ -515,7 +516,16 @@ export interface BaseExpectTypeOf<Actual, Options extends {positive: boolean}> {
    * @param args - The arguments to check for callability.
    * @returns `true`.
    */
-  toBeCallableWith: Options['positive'] extends true ? (...args: Params<Actual>) => true : never
+  toBeCallableWith: Options['positive'] extends true
+    ? <A extends OverloadParameters<Actual>>(
+        ...args: A
+      ) => {
+        parameters: ExpectTypeOf<FilterExtendedBy<OverloadParameters<Actual>, A>, Options>
+        parameter: <N extends number>(n: N) => ExpectTypeOf<FilterExtendedBy<OverloadParameters<Actual>, A>[N], Options>
```

Comment:
Sadly not, that was a what I tried first. That still allows any number input, even for tuples, so not sure there's a point. There might be a ts issue for it somewhere.

Reply:
Ok yeah it's that any number key is allowed, but string keys need are properly type-checked. We could consider making this a string type instead, and use template literals to only allow number-like strings, but I don't know if it's worth it...

<img width="171" alt="image" src="https://github.com/user-attachments/assets/e151e62c-0c14-4e4f-b909-c291700067c6">

Reply:
I'm going to leave this as is because it's not bad anyway - if you do `.parameter(1.5)` the follow-on assertion would fail (like `.parameter(0.5).toBeString()` - the only one that would pass would be `.parameter(0.5).toBeUndefined()` which matches runtime behaviour anyway.

If you can find a way to make it more strongly type-checked I'd be open to a follow-on PR but this one already is getting too big.